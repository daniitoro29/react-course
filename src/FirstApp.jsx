import PropTypes from 'prop-types'; // Para definir el tipo a la properties

export const FirstApp = ({title, subtitle, name}) => {


    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <p>{ subtitle}</p>
            <p>{ subtitle}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    title: "No hay titulo",
    subtitle: "No hay subtitle",
    name: "Daniela"
}
