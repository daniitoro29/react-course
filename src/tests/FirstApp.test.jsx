import {render} from '@testing-library/react';
import {FirstApp} from '../FirstApp';

describe('Pruebas firstApp', () => {
    test('Debe hacer match con el snapshot', () => {
        const title = 'title';
        const subtitle = 'hello';
        const nombre = 'Cris';
       const {container} = render(<FirstApp title={title}  subtitle={subtitle}  nombre={nombre}/>)
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el titulo en un h1', () => {
        const title = 'title';
        const subtitle = 'hello';
        const nombre = 'Cris';
       const {container, getByText, getByTestId} = render(<FirstApp title={title}  subtitle={subtitle}  nombre={nombre}/>)
       expect(getByText(title)).toBeTruthy();

        expect(getByTestId('test-title').innerHTML).toContain(title)
    });

    test('', () => {
        
    });
    
    
    
});
