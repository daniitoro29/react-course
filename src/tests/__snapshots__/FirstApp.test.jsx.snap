// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas firstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    title
  </h1>
  <p>
    hello
  </p>
  <p>
    hello
  </p>
</div>
`;
